src/components/TodoError.tsx

```diff
@@ -0,0 +1,47 @@
+import { useContext, useEffect, useState } from 'react';
+import { SetErrorContext } from '../utils/setErrorContext';
+
+interface Props {
+  error: string | null;
+}
+
+export const TodoError: React.FC<Props> = ({ error }) => {
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const setError = useContext(SetErrorContext);
+
+  useEffect(() => {
+    switch (error) {
+      case 'cantfetch':
+        setErrorMessage('Unable to fetch todos');
+        break;
+      case 'cantadd':
+        setErrorMessage('Unable to add a todo');
+        break;
+      case 'cantdelete':
+        setErrorMessage('Unable to delete a todo');
+        break;
+      case 'cantupdate':
+        setErrorMessage('Unable to update a todo');
+        break;
+      case 'emptytitle':
+        setErrorMessage('Title can\'t be empty');
+        break;
+      default:
```

Comment:
do you really need to create separate useEffect to handle all error message? it can be set once in finally block